s3fs/tests/test_s3fs.py

```diff
@@ -2464,6 +2465,34 @@ def test_exists_isdir(s3):
     assert not s3.isdir(bad_path)
 
 
+def test_exists_raises_on_connection_error(monkeypatch):
+    # Ensure that we raise a ConnectionError instead of returning False if we
+    # are not actually able to connect to storage, by setting a fake proxy and
+    # then re-creating the S3FileSystem instance.
+    monkeypatch.setenv('https_proxy', 'https://fakeproxy.127.0.0.1:8080')
+    S3FileSystem.clear_instance_cache()
+    s3 = S3FileSystem(anon=False, client_kwargs={"endpoint_url": endpoint_uri})
```

Comment:
`skip_instance_cache=True` so as not to affect future instances.
However, isn't it simpler to pass endpoint_url here?

Reply:
Not sure what you mean with simpler? However, I added `skip_instance_url` as requested.